Test against oldest versions of dependencies

Add oldest supported version tests. This is to check for failures in the tests, such as this one with old versions of numpy:

```
_________________________________________________________________ test_getdminfo[shape0] __________________________________________________________________
tmp_path = PosixPath('/tmp/pytest-of-ekoch/pytest-0/test_getdminfo_shape0_0'), shape = (3,)
    @pytest.mark.skipif('not CASATOOLS_INSTALLED')
    @pytest.mark.parametrize('shape', SHAPES)
    def test_getdminfo(tmp_path, shape):
        filename = str(tmp_path / 'test.image')
        data = np.random.random(shape)
        ia = image()
        ia.fromarray(outfile=filename, pixels=data, log=False)
        ia.close()
        tb = table()
        tb.open(filename)
        reference = tb.getdminfo()
        tb.close()
        actual = getdminfo(filename)
        # We include information about endian-ness in the dminfo but CASA doesn't
        actual['*1'].pop('BIGENDIAN')
        # The easiest way to compare the output is simply to compare the output
        # from pformat (checking for dictionary equality doesn't work because of
        # the Numpy arrays inside).
>       assert pformat(actual) == pformat(reference)
E       assert "{'*1': {'COL...dCellStMan'}}" == "{'*1': {'COLU...dCellStMan'}}"
E         Skipping 128 identical leading characters in diff, use -v to show
E         -  array([3]),
E         +  array([3], dtype=int32),
E                            'HYPERCUBES': {'*1': {'BucketSize': 12,
E         -                                        'CellShape': array([3]),
E         +                                        'CellShape': array([3], dtype=int32),
E         ?                                                              +++++++++++++...
E         
E         ...Full output truncated (12 lines hidden), use '-vv' to show
/home/ekoch/ownCloud/code_development/radio_astro_tools/casa-formats-io/casa_formats_io/tests/test_casa_low_level_io.py:51: AssertionError
```
 Ping @astrofrog

--- a/casa_formats_io/tests/test_casa_low_level_io.py
+++ b/casa_formats_io/tests/test_casa_low_level_io.py
@@ -47,8 +47,10 @@ def test_getdminfo(tmp_path, shape):
 
     # The easiest way to compare the output is simply to compare the output
     # from pformat (checking for dictionary equality doesn't work because of
-    # the Numpy arrays inside).
-    assert pformat(actual) == pformat(reference)
+    # the Numpy arrays inside). Older versions of casatools represent some of
+    # the vectors as int32 instead of native int but our implementation uses
+    # native int so strip any mention of int32 from reference output
+    assert pformat(actual) == pformat(reference).replace(', dtype=int32', '')
 
 
 def test_getdminfo_large():

--- a/casa_formats_io/tests/test_casa_low_level_io.py
+++ b/casa_formats_io/tests/test_casa_low_level_io.py
@@ -121,7 +121,10 @@ def test_generic_table_read(tmp_path):
     desc_reference = tb.getdesc()
     tb.close()
 
-    assert pformat(desc_actual) == pformat(desc_reference)
+    # Older versions of casatools represent some of the vectors as int32
+    # instead of native int but our implementation uses native int so strip any
+    # mention of int32 from reference output
+    assert pformat(desc_actual) == pformat(desc_reference).replace(', dtype=int32', '')
 
     # TODO: for now the following fails because we haven't implemented
     # non-tiled data I/O

--- a/tox.ini
+++ b/tox.ini
@@ -30,8 +30,8 @@ deps =
     devdeps: :NUMPY_NIGHTLY:numpy
     devdeps: pyerfa
     devdeps: :ASTROPY_NIGHTLY:astropy
-    casa: :NRAO:casatools
-    casa: :NRAO:casatasks
+    casa-oldestdeps: :NRAO:casatools==6.0.0.27
+    casa-!oldestdeps: :NRAO:casatools
 extras =
     test
 commands =

